File: samples/benchmarks/primes.c
// primes.c
// Implements a very naive prime calculation algorithm

#include <stdio.h>
#include <stdbool.h>

bool is_prime(unsigned int candidate);

int main(int argc, char* argv[]) {
    // calculate the first 1000 primes
    unsigned int i = 1;
    unsigned int count = 0;
    while (i < 1000) {
        if (is_prime(i)) {
            printf("%d, ", i);
        }
        
        i += 1;
    }
    printf("\n");
    printf("Found %d primes\n", count);

    return 0;
}

bool is_prime(unsigned int candidate) {
    // is_prime
    // determines whether the candidate is prime

    bool prime = true;  // change to false once we prove it's not prime
    if (candidate < 2) {
        prime = false;
    }
    else if (candidate == 2) {
        prime = true;
    }
    else {
        unsigned int i = 2;
        bool found_factor = false;
        while (i < candidate && !found_factor) {    // todo: use 
            if (candidate % i == 0) {
                found_factor = true;
                prime = false;
            }
            else {
                i++;
            }
        }
    }

    return prime;
}

File: samples/benchmarks/primes.py
# primes.py

def is_prime(candidate: int) -> bool:
    prime = True
    if candidate < 2:
        prime = False
    elif candidate > 2:
        i = 2
        while (i < candidate and prime):
            if candidate % i == 0:
                prime = False
            else:
                i += 1
    
    return prime

# main program
i = 1
count = 0

while i < 1000:
    if is_prime(i):
        count += 1
        print(i, ", ", sep="")
    i += 1

print()
print(f"Found {count} primes")
print("Done.")
